python/kserve/kserve/handlers/http.py

```diff
@@ -65,20 +66,68 @@ def get_model(self, name: str):
 
 
 class PredictHandler(HTTPHandler):
+    def create_cloudevent(self, model_name: str, body: Union[Dict, CloudEvent], response: Dict,
+                          binary_event=False) -> Dict:
+        ce_attributes = {}
+
+        if os.getenv("CE_MERGE", "false") == "true":
+            if binary_event:
+                ce_attributes = body._attributes
+                if "datacontenttype" in ce_attributes:
+                    del ce_attributes["datacontenttype"]
+            else:
+                ce_attributes = body
+                del ce_attributes["data"]
+
+            # Remove these fields so we generate new ones
+            del ce_attributes["id"]
+            del ce_attributes["time"]
+
+        ce_attributes["type"] = os.getenv("CE_TYPE", "io.kserve.inference.response")
+        ce_attributes["source"] = os.getenv("CE_SOURCE", f"io.kserve.kfserver.{model_name}")
+        event = CloudEvent(ce_attributes, response)
+
+        if binary_event:
+            event_headers, event_body = to_binary(event)
+        else:
+            event_headers, event_body = to_structured(event)
+
+        for k, v in event_headers.items():
+            if k == "ce-time":
+                # Set the time with the timezone data
+                self.set_header('ce-time', datetime.utcnow().replace(tzinfo=pytz.utc).
+                                strftime('%Y-%m-%dT%H:%M:%S.%f%z'))
+            else:
+                self.set_header(k, v)
+
+        self.set_header("Content-Type", "application/cloudevents+json")
```

Comment:
We should set the `Content-Type` header according to the Content Mode - binary or structured:

- For **binary** content mode we use `Content-Type: application/json; charset=utf-8` - the body contains only the data payload.
- For **structured** content mode we use `Content-Type: application/cloudevents+json; charset=UTF-8` - the body contains both the data payload and the CloudEvents attributes.

Reply:
Thanks. I was trying to find info like this before 👍